mlrun/model_monitoring/db/tsdb/timescaledb/writer_graph_steps.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2025 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from datetime import datetime
+
+import mlrun.common.schemas.model_monitoring as mm_schemas
+import mlrun.feature_store.steps
+from mlrun.utils import logger
+
+
+class ProcessBeforeTimescaleDBWriter(mlrun.feature_store.steps.MapClass):
+    def __init__(self, **kwargs):
+        """
+        Process the data before writing to TimescaleDB via the new async writer.
+
+        This step combines functionality from both the existing stream processor
+        and the TDEngine writer pattern to create appropriate table names and
+        format data for TimescaleDB writer targets.
+
+        :returns: Event as a dictionary which will be written into the TimescaleDB Metrics/App Results tables.
+        """
+        super().__init__(**kwargs)
```

Comment:
Redundant constructor. Docstring can move to class level.
```suggestion
        """
        Process the data before writing to TimescaleDB via the new async writer.
        This step combines functionality from both the existing stream processor
        and the TDEngine writer pattern to create appropriate table names and
        format data for TimescaleDB writer targets.
        :returns: Event as a dictionary which will be written into the TimescaleDB Metrics/App Results tables.
        """
```

Reply:
will fix